ui/shared/status/StatusChatInput.qml

```diff
@@ -871,11 +871,21 @@ Rectangle {
                 Keys.onReleased: onRelease(event) // gives much more up to date cursorPosition
                 Keys.onShortcutOverride: event.accepted = isUploadFilePressed(event)
                 leftPadding: 0
-                background: Rectangle {
-                    color: "transparent"
-                }
                 selectionColor: Style.current.primarySelectionColor
                 persistentSelection: true
+                onTextChanged: {
+                    var symbols = ":='xX><0O;*dB8-D#%\\";
+                    if (length > 1 && (symbols.indexOf(getText((length - 2), length - 1)) !== -1)) {
+                        const emojis = EmojiJSON.emoji_json.filter(function (emoji) {
```

Comment:
I think this way of parsing is a bit too complex and not super performant. 
Since you already have the `index` of the place where there is an emoji to add, you can maybe use some kind of Constant object to map the ascii emoji to the unicode one.
Something like:
```js
const ASCII_EMOJIS = {
  ":)": "unicode...",
  ":D": ...
}
```
You can then even use that object to parse the text with `Object.keys` to get all the ascii emojis. Like this:
```js
Object.keys(ASCII_EMOJIS).forEach(function (emoji, emojiIndex) {
  let text = getText(0, length - 1)
  const index = text.indexOf(emoji)
  if (index > -1) {
      replaceWithEmoji("", getText(index, index + ASCII_EMOJIS[emojiIndex].length), ASCII_EMOJIS[emojiIndex]);
  }
})
```

Also, I tested your change, and if you add an emoji (eg `:D`) in the middle of the text, it doesn't work. This change above **should** fix it.

Anyway, let me know if it makes sense. Maybe I'm wrong with some assumptions. Also, I didn't test the code, so there might be small bugs.

Reply:
H @jrainville, some emoji aliases have up to 6 symbol combinations (e.g. "aliases_ascii": [ "':)", "':-)", "'=)", "':D", "':-D", "'=D" ]), so do you suggest to add all these in ASCII_EMOJIS? Wouldn't this be a duplicate since the code here will start parsing the emojiList.js only if a symbol in the symbols list (which contains all possible initial symbols of all emoji aliases) is written in the text area, so depending on what the user types, could be for example "':)" or "':-)" or "'=)", we should get 1f605 unicode. Also not sure about maintaining 2 different lists with emojis, if there is any update we should take care and update all lists, would this be convenient? Regarding your suggestion will always be checking the text since the begging `let text = getText(0, length - 1)`, would that work with a text like ":D hello :D" or something similar? 

Regarding the issue you have that if you add an emoji in the middle of the text doesn't work, it's not happening for me (see video), could you tell me what you did exactly to see if I can reproduce it?


https://user-images.githubusercontent.com/31625338/125270333-3c913d80-e312-11eb-8bc7-42fe2611b359.mov

Reply:
Ah, I hadn't realized that we had that many aliases. I agree then that creating a second table is redundant.

I'm not sure what the best would be then. I guess after testing, the feature seems pretty snappy, so maybe it's not that big of a performance hit. I do have a good PC so it helps.

Anyway, here is how you can reproduce the issue of writing emojis in the middle:

https://user-images.githubusercontent.com/11926403/125306205-03020780-e2fd-11eb-8d3f-6793181542e1.mp4

I think it's just an issue of doing the getText with the cursor position, maybe?

Anyway, if you fix the issue, I think it's good

Reply:
Thanks a lot for the video @jrainville, please check on the new commit which I believe fixes the issue. Regarding the performance, @iurimatias mentioned that he noticed a performance drop in the component but it's already happening on master, not sure if there is already a bug for this.